main.py

```diff
@@ -36,41 +34,42 @@ def calculate_anisotropy_field():
     return -spins_array[atom].spin_position_z*spins_array[atom].spin_position_z*spins_array[atom].anisotropy
 
 
+if __name__ == '__main__':
 
-number_of_atoms = 1
-number_of_simulation_steps = 30
+    number_of_atoms = 1
+    number_of_simulation_steps = 30
+    timestep = 1
 
-spins_array=[0 for i in range(number_of_atoms)]
-atom = 0
-iron = material_properties(1,1,1)
-co = material_properties(2,2,2)
+    spins_array=[0 for i in range(number_of_atoms)]
+    atom = 0
+    iron = material(1,1,1)
+    co = material(2,2,2)
 
-spins_array[atom] = create_a_spin(0,0,0,0.4,0.8,0.1,"iron",1,1)
+    spins_array[atom] = create_a_spin(0,0,0,0.4,0.8,0.1,"iron",1,1)
```

Comment:
There are a lot of properties on the `create_a_spin` class. 

Could some of them be separated out from others, or grouped together in their own type?

Eg would it make sense to have some grouping of (x,y,z) coordinates then have two coordinate objects as properties of the spin object? Eg position and direction...?